src/Amazon.Kinesis.Firehose/KinesisFirehoseClient.cs

```diff
@@ -87,6 +105,32 @@ private HttpRequestMessage GetRequestMessage<T>(string action, T request)
             };
         }
 
+        public static string ConvertBatchRequestToJsonManually<T>(T request)
+        {
+            string json;
+            var batchRequest = request as PutRecordBatchRequest;
+
+            StringBuilder jsonBuilder = new StringBuilder();
+            jsonBuilder.Append(@"{""DeliveryStreamName"":""");
+            jsonBuilder.Append(batchRequest.DeliveryStreamName);
+            jsonBuilder.Append(@""",""Records"":[");
+            for (int i = 0; i < batchRequest.Records.Length; i++)
+            {
+                var record = batchRequest.Records[i];
+
+                if (i > 0)
+                {
+                    jsonBuilder.Append(",");
+                }
+                jsonBuilder.Append(@"{""Data"":""");
+                jsonBuilder.Append(Convert.ToBase64String(record.Data));
+                jsonBuilder.Append(@"""}");
+            }
+            jsonBuilder.Append("]}");
+            json = jsonBuilder.ToString();
+            return json;
+        }

```

Comment:
This is what worked... So long as the stream name is plain ASCII without quotes, this shouldn't be in danger of breaking. It's just Base64 binary record data and a `DeliveryStreamName`.